picolibc.ld.in

```diff
@@ -283,6 +286,26 @@ SECTIONS
 		. = . + SIZEOF(.tbss);
 	} >ram AT>ram :ram
 	@BFD_END@
+	/*
```

Comment:
This needs to be in the .bss block so that the various addresses and sizes are computed correctly. It should be sufficient to just move the contents of this block to the top of the .bss section; all of the attributes are the same.

Reply:
Hrm. Perhaps you could do something similar to how the stack check block is allocated for powerpc -- the allocation in picolibc.ld only ensures there is sufficient space, it isn't the actual location of the block due to alignment and padding. The actual address is computed at runtime based off other symbols. It makes the code messier, but avoids having extra sections in the output which will confuse the runtime setup code.

Check out nelwib/libc/machine/powerpc/machine/_ssp_tls.h for the details on this hack.